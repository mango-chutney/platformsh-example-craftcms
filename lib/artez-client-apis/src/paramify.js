/*
 * paramify.js
 *
 * @flow
 */

export default function(params: Object): string {
  return Object.keys(params)
    .filter(key => params[key] !== undefined)
    .map(
      key =>
        params[key] === undefined
          ? ''
          : `${key}=${encodeURIComponent(params[key])}`,
    )
    .join('&');
}
