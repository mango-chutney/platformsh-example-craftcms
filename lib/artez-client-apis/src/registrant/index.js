/*
 * registrant/index.js
 *
 * @flow
 */

import personalizationPost from './personalizationPost';
import type { Config } from '../config';

export { personalizationPost };

export function bind(config: Config): * {
  return {
    personalizationPost: (...args: *) => personalizationPost(config, ...args),
  };
}

export const bindWithBaseURL = bind;
