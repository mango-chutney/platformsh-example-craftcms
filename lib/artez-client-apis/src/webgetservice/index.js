/*
 * webgetservice/index.js
 *
 * @flow
 */

import getEventFundraisingTotals from './getEventFundraisingTotals';
import getParticipantScoreBoard from './getParticipantScoreBoard';
import getRegistrant from './getRegistrant';
import getTeamScoreBoard from './getTeamScoreBoard';
import mobileParticipant from './mobileParticipant';
import postMessageBoardMain from './postMessageBoardMain';
import teamByTeamID from './teamByTeamID';
import teamSearchReg from './teamSearchReg';
import type { Config } from '../config';

export { getEventFundraisingTotals };
export { getParticipantScoreBoard };
export { getRegistrant };
export { getTeamScoreBoard };
export { mobileParticipant };
export { postMessageBoardMain };
export { teamByTeamID };
export { teamSearchReg };

export function bind(config: Config): * {
  const { baseURL } = config;
  if (baseURL.match(/get[.]asmx\/?$/)) {
    throw new Error('Do not postfix baseURL with get.asmx');
  }

  if (baseURL.match(/mobile[.]asmx\/?$/)) {
    throw new Error('Do not postfix baseURL with mobile.asmx');
  }

  return {
    getEventFundraisingTotals: (...args: *) =>
      getEventFundraisingTotals(config, ...args),
    getParticipantScoreBoard: (...args: *) =>
      getParticipantScoreBoard(config, ...args),
    getRegistrant: (...args: *) => getRegistrant(config, ...args),
    getTeamScoreBoard: (...args: *) => getTeamScoreBoard(config, ...args),
    mobileParticipant: (...args: *) => mobileParticipant(config, ...args),
    postMessageBoardMain: (...args: *) => postMessageBoardMain(config, ...args),
    teamByTeamID: (...args: *) => teamByTeamID(config, ...args),
    teamSearchReg: (...args: *) => teamSearchReg(config, ...args),
  };
}

export const bindWithBaseURL = bind;
