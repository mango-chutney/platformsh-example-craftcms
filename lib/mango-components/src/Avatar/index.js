// @flow

export { default as Avatar } from './Avatar';
export { default as AvatarComponent } from './AvatarComponent';
export { default as AvatarComposer } from './AvatarComposer';
