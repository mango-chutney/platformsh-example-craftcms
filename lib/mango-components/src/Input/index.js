// @flow

export { default as createInputIdAttribute } from './createInputIdAttribute';
export { default as createLabelForAttribute } from './createLabelForAttribute';
export { default as createLabelObject } from './createLabelObject';
export { default as createLabelProps } from './createLabelProps';
export { default as defaultTheme } from './styles';
export { default as Input } from './Input';
export { default as InputComponent } from './InputComponent';
export { default as InputComposer } from './InputComposer';
export { default as InputDecoratorComponent } from './InputDecoratorComponent';
export { default as InputLabelComponent } from './InputLabelComponent';
export {
  default as createFormControlElementProps,
} from './createFormControlElementProps';
export {
  default as createInputDecoratorProps,
} from './createInputDecoratorProps';
