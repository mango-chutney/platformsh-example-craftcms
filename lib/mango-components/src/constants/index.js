import { rem } from 'polished';

export const palette = {
  primary: '#2ea1f8',
  secondary: '#1bb934',
  tertiary: '#222c3c',
  lightGray: '#fafbfc',
  mediumGray: '#cacaca',
  darkGray: '#7f8fa4',
  white: '#fefefe',
  black: '#354052',
  body: '#7f8fa4',
  heading: '#354052',
  subheading: '#7f8fa4',
  border: '#e2e7ee',
  success: '#00D161',
  alert: '#F85359',
  inactive: '#C5D0DE',
};

export const fontWeights = {
  light: 300,
  normal: 'normal',
  medium: 500,
  semibold: 600,
  bold: 800,
  extrabold: 900,
};

export const fontStack =
  'proxima-nova,Helvetica Neue Light,Helvetica Neue,Helvetica,Arial,sans-serif';

export const globalRadius = rem(4);

export const globalBorder = `${rem(1)} solid ${palette.border}`;
