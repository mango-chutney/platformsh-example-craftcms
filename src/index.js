// @flow

export { default as AnimatedRainbowGradient } from './AnimatedRainbowGradient';

export { Avatar, AvatarComponent, AvatarComposer } from './Avatar';

export {
  AnchorButton,
  AnchorButtonComponent,
  Button,
  ButtonComponent,
  ButtonComposer,
} from './Button';

export {
  Checkbox,
  CheckboxComposer,
  CheckboxContainerComponent,
  CheckboxInputComponent,
  CheckboxInputDecoratorComponent,
  CheckboxLabelComponent,
  CheckboxWrapperComponent,
} from './Checkbox';

export {
  CoolChild,
  CoolChildrenWrapperComponent,
  Cooltip,
  CooltipComposer,
  CooltipWrapperComponent,
  OtherChild,
  PipComponent,
  TiptextComponent,
  TiptextWrapperComponent,
} from './Cooltip';

export {
  DatePicker,
  DatePickerComposedInputComponent,
  DatePickerComposer,
  DatePickerOverlayComponent,
  DatePickerOverlayWrapperComponent,
} from './DatePicker';

export {
  Input,
  InputComponent,
  InputComposer,
  InputDecoratorComponent,
  InputLabelComponent,
  createFormControlElementProps,
  createInputDecoratorProps,
  createInputIdAttribute,
  createLabelForAttribute,
  createLabelObject,
  createLabelProps,
} from './Input';

export {
  MadeWithMangoChutney,
  MadeWithMangoChutneyAnchorComponent,
  MadeWithMangoChutneyComponent,
  MadeWithMangoChutneyWrapperComponent,
} from './MadeWithMangoChutney';

export { Panel, PanelHeading, PanelSection } from './Panel';

export {
  Radio,
  RadioComposer,
  RadioInputComponent,
  RadioInputDecoratorComponent,
  RadioLabelComponent,
  RadioWrapperComponent,
} from './Radio';

export {
  Select,
  SelectComponent,
  SelectComposer,
  SelectInputDecoratorComponent,
  SelectLabelComponent,
} from './Select';

export {
  TextArea,
  TextAreaComponent,
  TextAreaComposer,
  TextAreaInputDecoratorComponent,
  TextAreaLabelComponent,
} from './TextArea';

export {
  PersistentToast,
  PersistentToastComposer,
  RainbowToastComponent,
  Toast,
  ToastComponent,
  ToastComposer,
  ToastConsumer,
  ToastContainerComponent,
  ToastPortal,
  ToastPortalComposer,
  ToastProvider,
  transitionStates,
} from './Toast';

export {
  TootlipArrowComponent,
  TootlipPopperComponent,
  Tootlip,
  UnmanagedTootlip,
  UnmanagedTootlipComposer,
  placements as tootlipPlacements,
} from './Tootlip';

export {
  Tristicon,
  TristiconComponent,
  TristiconComposer,
  createTristiconFontFace,
  tristicons,
} from './Tristicon';

export {
  TypeaheadInput,
  TypeaheadInputComponent,
  TypeaheadInputComposer,
  TypeaheadItemComponent,
  TypeaheadMenuComponent,
  TypeaheadMenuWrapperComponent,
} from './TypeaheadInput';
