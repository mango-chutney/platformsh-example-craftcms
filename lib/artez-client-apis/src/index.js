/*
 * index.js
 *
 * @flow
 */

import * as regapi from './regapi';
import * as registrant from './registrant';
import * as webgetservice from './webgetservice';

export { regapi, registrant, webgetservice };

export default { regapi, registrant, webgetservice };
