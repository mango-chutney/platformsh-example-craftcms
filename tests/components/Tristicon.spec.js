// @flow

import 'raf/polyfill';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { Tristicon } from '../../src';

function ExampleTristicon() {
  return <Tristicon icon="star" />;
}

it('Tristicon component renders correctly', () => {
  const tree = renderer.create(<ExampleTristicon />).toJSON();
  expect(tree).toMatchSnapshot();
});
