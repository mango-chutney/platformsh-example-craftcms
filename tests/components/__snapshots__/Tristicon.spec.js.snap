// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tristicon component renders correctly 1`] = `
.c0::before {
  content: "\\f13b";
  display: inline-block;
  font: normal normal normal 0.875rem tristicons;
  font-size: inherit;
  text-rendering: auto;
  vertical-align: inherit;
}

<i
  className="c0"
  icon="star"
/>
`;
